### DISCUSSION TITLE
Modifying varaible resolving

### DISCUSSION THREAD

**Discussion by shubhamshirude2512** [ksh93/ksh#645] [Q&A] [answered]
Hello,
I want to modify the variable resolution

for example

if [[ "a" -eq $x ]]
then
echo "in if block
fi

Now the variable x is not not defined so echo $x prints nothing

but still should execute the if block.

I know this sounds wierd but I was just wondering if we could alter variable assignment in ksh source code.

Thanks in advance

**Comment by McDutchie** [comment, marked as answer]
ksh does not need modifying here, your script does. For one thing, using `-eq` is wrong there as that compares numerical values or arithmetic expressions; you want `==` for string (pattern) comparison.

So, if you want the `if` block to be executed if either $x is "a" or x is not defined (a.k.a. unset), you want:

```sh
if [[ $x == "a" || ! -v x ]]
then
     echo "in if block"
fi
```

The `-v` operator to `[[` checks if the variable is set. The `!` negates it so the condition is true if the variable is unset. If you want to check whether it is unset or set to empty, then you need `-z` instead. See the manual page for more details.

Hope this helps.

**Reply by shubhamshirude2512** [reply]
Hello, I dont want to modify the script,
I just never want any variable's value to be "" whenever I compare it in IF block.

it needs to be 

for example

if [[ -z "$var" ]]
then
      echo "\$var is empty"
else
      echo "\$var is NOT empty"
fi


this should always print "var is NOT empty" even when var is not assigned


I dont want to modify the script,
i want to make it the default behaviour of my custom ksh

**Reply by McDutchie** [reply]
This makes no sense, and it's not a change I'm going to make. You're on your own.

**Reply by shubhamshirude2512** [reply]
hehe, thanks
